### PR TITLE
[3216] Write UI comp ViewModel tests

### LINKED ISSUE #3216: Port existing tests for UI Components ViewModels after migration to `OfflinePlugin`
### Context & proposal

Currently the tests for our UI Components ViewModels rely on `ChatDomain` and have been rendered inoperable. It is required to write tests for the new logic which relies on `OfflinePlugin`

### Acceptance criteria

To accept this task we need to complete the following items:

- [ ] Port existing tests to Offline plugin and make sure they run reliably

### PR DESCRIPTION
**Pull request by MarinTolic**
### 🎯 Goal

closes #3216 

### 🛠 Implementation details

Wrote tests for the ViewModels used in UI Components

#### General
- [ ] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [ ] PR targets the `develop` branch
- [ ] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### 🎉 GIF

gif

### CODE REVIEW

File: stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+                )
+                .givenChannelMutes()
+                .givenIsOffline(false)
+                .get()
+
+            val mockObserver: Observer<ChannelListViewModel.State> = mock()
+            viewModel.state.observeForever(mockObserver)
+
+            viewModel.onAction(ChannelListViewModel.Action.ReachedEndOfList)
+
```
Comment (skydoves) [inline stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt:201]: What do you think about adding `removeObserver` with some other verifying below?

```kotlin
val state = ChannelListViewModel.State(isLoading = false, channels = listOf(channel1, channel2))
verify(mockObserver, atLeastOnce()).onChanged(state)
viewModel.state.removeObserver(mockObserver)
```

File: stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+                )
+                .givenChannelMutes()
+                .givenIsOffline(true)
+                .get()
+
+            val mockObserver: Observer<ChannelListViewModel.State> = mock()
+            viewModel.state.observeForever(mockObserver)
+
+            viewModel.onAction(ChannelListViewModel.Action.ReachedEndOfList)
+
```
Comment (skydoves) [inline stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt:228]: Also, we can add the `removeObserver` here for avoiding unexpected behavior.

**Review by skydoves — COMMENTED** [review]
Overall looks nice! I left very trivial comments below. 😃

File: stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt (hunk 1 above)
Comment (MarinTolic) [reply stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt:201]: Good suggestion.

Removed the observer at the end of the test.

File: stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt (hunk 2 above)
Comment (MarinTolic) [reply stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt:230]: Good suggestion.

Removed the observer at the end of the test.

**Review by bychkovdmitry — CHANGES_REQUESTED** [review]

**Review by bychkovdmitry — APPROVED** [review]
LGTM!

File: stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,339 @@
...
+import org.mockito.kotlin.times
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+
+@OptIn(ExperimentalCoroutinesApi::class)
+internal class ChannelListViewModelTest {
+
+    @JvmField
+    @RegisterExtension
+    val instantExecutorExtension: InstantExecutorExtension = InstantExecutorExtension()
```
[the remark this answers is not in the document]
Comment (MarinTolic) [reply stream-chat-android-ui-components/src/test/kotlin/io/getstream/chat/android/ui/viewmodels/ChannelListViewModelTest.kt]: Updated to use it.

**Pull request merged**